### DISCUSSION TITLE
Spancat scripted training without CLI commands

### DISCUSSION THREAD

**Discussion by harshchavhan** [explosion/spaCy#11145] [Help: Other Questions]
facing issues in passing annotated jsonl format training-dataset file which is generated by prodigy db-out command. 
Trying to train prodigy annotate dataset directly on spacy without using CLI commands @polm

**Comment by ljvmiranda921** [comment]
Hi @harshchavhan ,

We appreciate not tagging maintainers directly in the discussion forum. If you already exported your data from Prodigy using [`db-out`](https://prodi.gy/docs/recipes#db-out), then you can use the [`data-to-spacy`](https://prodi.gy/docs/recipes#data-to-spacy) command to convert the data into the [spaCy format](https://spacy.io/api/data-formats#binary-training). From there, you can use [`spacy train`](https://spacy.io/api/cli#train) to perform training. You can check the [training documentation](https://spacy.io/usage/training) for more info.

> facing issues in passing annotated jsonl format training-dataset file

Can you copy-paste the text of the error or issue?

It's a bit unclear what you meant by not using CLI commands. If you want to perform training programmatically, one way is to see [how it's done in `spacy train`](https://github.com/explosion/spaCy/blob/master/spacy/cli/train.py) and adapt it into your codebase. Under the hood, it calls the [`train` function ](https://github.com/explosion/spaCy/blob/master/spacy/training/loop.py#L24) for model training.
